### PR TITLE
[v6] Avoid password and resetKey column to be sorted by the /users API

### PR DESCRIPTION
**Pull request by avdev4j**
cc @pascalgrimaud 
---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/src/main/java/package/web/rest/UserResource.java.ejs [hunk 1]
```diff
@@ -286,18 +292,34 @@ public class UserResource {
     <%_ if (databaseType === 'sql' || databaseType === 'mongodb' || databaseType === 'neo4j' || databaseType === 'couchbase') { _%>
         <%_ if (!reactive) { _%>
     public ResponseEntity<List<<%= asDto('User') %>>> getAllUsers(Pageable pageable) {
+        <%_ if (authenticationType !== 'oauth2') { _%>
+        if (!onlyContainsAllowedProperties(pageable)) {
+            return ResponseEntity.badRequest().build();
+        }
+        <% } %>
```
Comment (pascalgrimaud) [inline generators/server/templates/src/main/java/package/web/rest/UserResource.java.ejs:299]: should be changed to `<%_ } _%>`

File: generators/server/templates/src/main/java/package/web/rest/UserResource.java.ejs [hunk 2]
```diff
@@ -286,18 +292,34 @@ public class UserResource {
...
         final Page<<%= asDto('User') %>> page = userService.getAllManagedUsers(pageable);
         HttpHeaders headers = PaginationUtil.generatePaginationHttpHeaders(ServletUriComponentsBuilder.fromCurrentRequest(), page);
         return new ResponseEntity<>(page.getContent(), headers, HttpStatus.OK);
         <%_ } else { _%>
     public Mono<ResponseEntity<Flux<<%= asDto('User') %>>>> getAllUsers(ServerHttpRequest request, Pageable pageable) {
+        <%_ if (authenticationType !== 'oauth2') { _%>
+        if (!onlyContainsAllowedProperties(pageable)) {
+            return Mono.just(ResponseEntity.badRequest().build());
+        }
+        <% } %>
```
Comment (pascalgrimaud) [inline generators/server/templates/src/main/java/package/web/rest/UserResource.java.ejs:309]: should be changed to `<%_ } _%>`

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

File: generators/server/templates/src/main/java/package/web/rest/UserResource.java.ejs (hunk 1 above)
Comment (avdev4j) [reply generators/server/templates/src/main/java/package/web/rest/UserResource.java.ejs:299]: It's intended to let a blank line between the end of the if statement and the rest of the code.

File: generators/server/templates/src/main/java/package/web/rest/UserResource.java.ejs (hunk 2 above)
Comment (avdev4j) [reply generators/server/templates/src/main/java/package/web/rest/UserResource.java.ejs:309]: same

File: generators/server/templates/src/main/java/package/web/rest/UserResource.java.ejs (hunk 1 above)
Comment (DanielFran) [reply generators/server/templates/src/main/java/package/web/rest/UserResource.java.ejs:299]: Better to use always same syntax <%_ } _%> and explicitly add a blank line

File: generators/server/templates/src/main/java/package/web/rest/UserResource.java.ejs (hunk 2 above)
Comment (DanielFran) [reply generators/server/templates/src/main/java/package/web/rest/UserResource.java.ejs:309]: Better to use always same syntax <%_ } _%> and explicitly add a blank line

**Review by DanielFran — CHANGES_REQUESTED** [review]

File: generators/server/templates/src/main/java/package/web/rest/UserResource.java.ejs (hunk 2 above)
Comment (avdev4j) [reply generators/server/templates/src/main/java/package/web/rest/UserResource.java.ejs]: right ;)

**Comment by pascalgrimaud** [comment]
@avdev4j : all webflux builds failed. Can you have a look plz ?

**Pull request merged**